charts/docker-minio/templates/ingress.yaml

```diff
@@ -0,0 +1,61 @@
+{{- if .Values.ingress.enabled -}}
+{{- $fullName := include "docker-minio.fullname" . -}}
+{{- $svcPort := .Values.service.port -}}
+{{- if and .Values.ingress.className (not (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion)) }}
+  {{- if not (hasKey .Values.ingress.annotations "kubernetes.io/ingress.class") }}
+  {{- $_ := set .Values.ingress.annotations "kubernetes.io/ingress.class" .Values.ingress.className}}
+  {{- end }}
+{{- end }}
+{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1
+{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1beta1
+{{- else -}}
+apiVersion: extensions/v1beta1
+{{- end }}
+kind: Ingress
+metadata:
+  name: {{ $fullName }}
+  labels:
+    {{- include "docker-minio.labels" . | nindent 4 }}
+  {{- with .Values.ingress.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  {{- if and .Values.ingress.className (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion) }}
+  ingressClassName: {{ .Values.ingress.className }}
+  {{- end }}
+  {{- if .Values.ingress.tls }}
+  tls:
+    {{- range .Values.ingress.tls }}
+    - hosts:
+        {{- range .hosts }}
+        - {{ . | quote }}
+        {{- end }}
+      secretName: {{ .secretName }}
+    {{- end }}
+  {{- end }}
+  rules:
+    {{- range .Values.ingress.hosts }}
+    - host: {{ .host | quote }}
+      http:
+        paths:
+          {{- range .paths }}
+          - path: {{ .path }}
+            {{- if and .pathType (semverCompare ">=1.18-0" $.Capabilities.KubeVersion.GitVersion) }}
+            pathType: {{ .pathType }}
+            {{- end }}
+            backend:
+              {{- if semverCompare ">=1.19-0" $.Capabilities.KubeVersion.GitVersion }}
+              service:
+                name: {{ $fullName }}
+                port:
+                  number: {{ $svcPort }}
+              {{- else }}
+              serviceName: {{ $fullName }}
+              servicePort: {{ $svcPort }}
+              {{- end }}
+          {{- end }}
+    {{- end }}
+{{- end }}
```

Comment:
actually, I am not quite sure how useful can be to expose the console ingress, my idea is to have just the svc/minio-console and if/when you need it, you can proxy the port on your local host. I would avoid to publicly expose the console, is too risky.

Reply:
True, the only exposed port is the 9000:

- https://github.com/sparkfabrik/helm-charts/pull/5/files#diff-215428ba09929eebb0c84ae1c509bf42252a72c2870f829840a554cc99bd7e16R58
- https://github.com/sparkfabrik/helm-charts/pull/5/files#diff-f0d3512467f7599d5bcd74c5da829bc9804310d6548c7594a23c5ddf62b66dcbR3
- https://github.com/sparkfabrik/helm-charts/pull/5/files#diff-f0d3512467f7599d5bcd74c5da829bc9804310d6548c7594a23c5ddf62b66dcbR54

If the ingress is enabled, the only service port mapped is the port declared at `service.port`. The `console.enabled` only create another port in the service and it will be reachable only using `kubectl port-forward`.

The following YAML is a test manifest generated with `console.enabled` configured as `true`:

```yaml
---
# Source: docker-minio/templates/ingress.yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: docker-minio-test
  labels:
    helm.sh/chart: docker-minio-0.1.0
    app.kubernetes.io/name: docker-minio
    app.kubernetes.io/instance: docker-minio-test
    app.kubernetes.io/version: "latest"
    app.kubernetes.io/managed-by: Helm
spec:
  rules:
    - host: "chart-example.local"
      http:
        paths:
          - path: /
            pathType: ImplementationSpecific
            backend:
              service:
                name: docker-minio-test
                port:
                  number: 9000
```

Reply:
Ok as discussed, we can proceed with a new configuration value `ingressEnabled` and a custom host.